src/main/java/org/ow2/proactive/catalog/repository/BucketRepository.java

```diff
@@ -62,26 +61,38 @@
     BucketEntity findBucketForUpdate(String bucketName);
 
     @Query(value = "SELECT bk.bucketName, bk.owner, COUNT(cos.id.name) as objectCount, bk.id FROM BucketEntity bk LEFT JOIN bk.catalogObjects cos" +
-                   " WHERE lower(cos.kind) LIKE lower(concat(?1, '%')) AND lower(cos.contentType) LIKE lower(concat(?2, '%'))" +
+                   " WHERE lower(cos.kind) IN ?1 AND lower(cos.contentType) LIKE lower(concat(?2, '%'))" +
                    " AND lower(cos.id.name) LIKE lower(concat('%', ?3, '%')) OR bk.catalogObjects IS EMPTY GROUP BY bk.bucketName, bk.owner, bk.id")
-    List<Object[]> findContainingKindAndContentTypeAndObjectName(String kind, String contentType, String objectName,
-            Sort sort);
+    List<Object[]> findContainingKindAndContentTypeAndObjectName(List<String> kindList, String contentType,
+            String objectName, Sort sort);
 
     @Query(value = "SELECT bk.bucketName, bk.owner, COUNT(cos.id.name) as objectCount, bk.id FROM BucketEntity bk LEFT JOIN bk.catalogObjects cos" +
-                   " WHERE lower(cos.kind) LIKE lower(concat(?1, '%')) " +
+                   " WHERE lower(cos.contentType) LIKE lower(concat(?1, '%'))" +
+                   " AND lower(cos.id.name) LIKE lower(concat('%', ?2, '%')) OR bk.catalogObjects IS EMPTY GROUP BY bk.bucketName, bk.owner, bk.id")
+    List<Object[]> findContainingContentTypeAndObjectName(String contentType, String objectName, Sort sort);
+
+    @Query(value = "SELECT bk.bucketName, bk.owner, COUNT(cos.id.name) as objectCount, bk.id FROM BucketEntity bk LEFT JOIN bk.catalogObjects cos" +
+                   " WHERE lower(cos.kind) IN ?1 " +
                    " OR bk.catalogObjects IS EMPTY GROUP BY bk.bucketName, bk.owner, bk.id")
-    List<Object[]> findContainingKind(String kind, Sort sort);
+    List<Object[]> findContainingKind(List<String> kindList, Sort sort);
 
     @Query(value = "SELECT bk.bucketName, bk.owner, COUNT(cos.id.name) as objectCount, bk.id FROM BucketEntity bk LEFT JOIN bk.catalogObjects cos" +
-                   " WHERE bk.owner in ?1 AND (lower(cos.kind) LIKE lower(concat(?2, '%'))" +
+                   " WHERE bk.owner in ?1 AND (lower(cos.kind) IN ?2" +
```

Comment:
this induces a behavior change.

In the catalog portal, you can search using kind=workflow, it will find objects which are workflow/standard and workflow/psa. Removing the LIKE prevents to do this.